library/src/scala/tasty/reflect/Core.scala

```diff
@@ -190,8 +258,55 @@ trait Core {
     /** Type tree representing a type written in the source */
     type TypeTree <: TypeOrBoundsTree
 
-      // TODO Add subtype types of TypeTree for documentation? Or support refined bindings and add the types.
+    /** Type trees representing a type written in the source */
+    val TypeTree: TypeTreeCoreModule
+
+    /** Type trees representing a type written in the source */
+    abstract class TypeTreeCoreModule {
+
+      /** Type tree representing an inferred type */
+      type Synthetic <: TypeTree
```

Comment:
`Inferred` instead of `Synthetic`?

Reply:
Good idea. I will propose it in another PR.